tileview/src/main/java/com/qozix/tileview/tiles/TileCanvasViewGroup.java

```diff
@@ -31,6 +31,7 @@
   private TileRenderTask mLastRunTileRenderTask;
 
   private DetailLevel mDetailLevelToRender;
+  private DetailLevel mLastRequestedDetailLevel;
```

Comment:
While i allowed a zoom to trigger `DetailLevel` changes (just in case of a double-tap), i finally understood what caused #252. I warn you : this is complicated.

So you need a zoom that will cause a render of a lot of tiles. Double-tap from max scale is perfect. You need a map with enough `DetailLevel`. The demo may be enough. To emphasize this, set `TileView.setShouldScaleToFit( false )` and have a minimum scale equal or below the minimum `DetailLevel`'s scale to be be able to see every levels (and trigger the issue). So from max scale you double-tap. What happens next? Because of the lock, the `DetailLevel` don't change until the zoom ends. But before it ends, numerous tiles from that level will be rendered and scaled because during the zoom, `TileView.onScrollChanged` is being called (which indirectly calls `requestRender`). That's were we have a likelyhood of OOM. When the zoom ends, it removes the lock, updates the scale of the `DetailLevelManager`, which in turn calls `DetailLevelManager.update()` and then triggers a `DetailLevel` change. Given the impelmentation of a `DetailLevel` change in `TileView` :
```
@Override
public void onDetailLevelChanged( DetailLevel detailLevel ) {
   requestRender();
   mTileCanvasViewGroup.updateTileSet( detailLevel );
  }
```
the updateTileSet calls a `cancelRender`. So it attempts to stop the ongoing task mentioned before (which takes some time because of the number of the tiles to render). At this point we may say : "that's ok, after that a `requestRender` is called and everything will be fine". BUT, it turn out, that request may just have a `DetailLevel.computeCurrentState()` returning false, so the `requestRender` does nothing. So why `computeCurrentState` may return false? While zooming, and ongoing `requestRender` may call `computeCurrentState` at a point were the current `DetailLevel` is equal to the "destination" one (idk how to say this :). So, in `DetailLevel`, the attribute `mLastStateSnapshot` already has its end value, and so as the other parameters `rowStart`, `columnStart`, ...
And there we have it : the requestRender that does nothing! :)